examples/filter/pipeline_passthru/main/passthru.c

```diff
@@ -0,0 +1,115 @@
+/* Audio passthru
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+
+#include <string.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+#include "audio_pipeline.h"
+#include "i2s_stream.h"
+#include "esp_peripherals.h"
+#include "audio_hal.h"
+
+static const char *TAG = "PASSTHRU";
+
+void app_main(void)
+{
+    audio_pipeline_handle_t pipeline;
+    audio_element_handle_t i2s_stream_writer, i2s_stream_reader;
+
+    esp_log_level_set("*", ESP_LOG_INFO);
+    esp_log_level_set(TAG, ESP_LOG_DEBUG);
+
+    ESP_LOGI(TAG, "[ 1 ] Start codec chip");
+    audio_hal_codec_config_t audio_hal_codec_cfg =  AUDIO_HAL_ES8388_DEFAULT();
+    audio_hal_codec_cfg.adc_input = AUDIO_HAL_ADC_INPUT_LINE2;
+    audio_hal_codec_cfg.i2s_iface.samples = AUDIO_HAL_44K_SAMPLES;
+    audio_hal_handle_t hal = audio_hal_init(&audio_hal_codec_cfg, 0);
+    audio_hal_ctrl_codec(hal, AUDIO_HAL_CODEC_MODE_BOTH, AUDIO_HAL_CTRL_START);
+
+    ESP_LOGI(TAG, "[ 2 ] Create audio pipeline for playback");
+    audio_pipeline_cfg_t pipeline_cfg = DEFAULT_AUDIO_PIPELINE_CONFIG();
+    pipeline = audio_pipeline_init(&pipeline_cfg);
+
+    ESP_LOGI(TAG, "[3.1] Create i2s stream to write data to codec chip");
+    i2s_stream_cfg_t i2s_cfg = I2S_STREAM_CFG_DEFAULT();
+    i2s_cfg.type = AUDIO_STREAM_WRITER;
+    i2s_stream_writer = i2s_stream_init(&i2s_cfg);
+
+    ESP_LOGI(TAG, "[3.2] Create i2s stream to read data from codec chip");
+    i2s_stream_cfg_t i2s_cfg_read = I2S_STREAM_CFG_DEFAULT();
+    i2s_cfg_read.type = AUDIO_STREAM_READER;
+    i2s_stream_reader = i2s_stream_init(&i2s_cfg_read);
+
+    ESP_LOGI(TAG, "[3.3] Register all elements to audio pipeline");
+    audio_pipeline_register(pipeline, i2s_stream_reader, "i2s_read");
+    audio_pipeline_register(pipeline, i2s_stream_writer, "i2s_write");
+
+    ESP_LOGI(TAG, "[3.4] Link it together [codec_chip]-->i2s_stream_reader-->i2s_stream_writer-->[codec_chip]");
+    audio_pipeline_link(pipeline, (const char *[]) {"i2s_read", "i2s_write"}, 2);
+
+    ESP_LOGI(TAG, "[ 4 ] Initialize peripherals");
+    esp_periph_config_t periph_cfg = { 0 };
+    esp_periph_init(&periph_cfg);
+
+    ESP_LOGI(TAG, "[ 5 ] Setup event listener");
+    audio_event_iface_cfg_t evt_cfg = AUDIO_EVENT_IFACE_DEFAULT_CFG();
+    audio_event_iface_handle_t evt = audio_event_iface_init(&evt_cfg);
+
+    ESP_LOGI(TAG, "[5.1] Listening event from all elements of pipeline");
+    audio_pipeline_set_listener(pipeline, evt);
+
+    ESP_LOGI(TAG, "[5.2] Listening event from peripherals");
+    audio_event_iface_set_listener(esp_periph_get_event_iface(), evt);
```

Comment:
Consequently step "[5.2] Listening event from peripherals: is not required as well